pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableImplV4.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.core.common.datatable;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.core.common.datablock.RowDataBlock;
+import org.apache.pinot.spi.annotations.InterfaceStability;
+
+
+/**
+ * Datatable V4 Implementation is a wrapper around the Row-based data block.
+ */
+@InterfaceStability.Evolving
+public class DataTableImplV4 extends RowDataBlock {
```

Comment:
This PR https://github.com/apache/pinot/pull/8872/ is also implementing `DataImplV4` for a different reason though (null handling feature in the existing engine).

We should name this differently since DataImplV1 / V2 / V3 is the convention being followed in the existing engine's data table transport

Reply:
Also I wonder why is this subclassing `RowDataBlock` and not `BaseDataBlock` because otherwise this will be specific only to row oriented format ?

Reply:
`RowDataBlock` is in fact a fully compatible `DataTableImplV4` with row-major storage byte buffer format. planning to reuse this going forward.

Reply:
Discussed this offline with @walterddr  and clarified. This PR makes in particular has got nothing to do with multi-stage engine. The goal of this PR is to leverage the RowDataBlock (which was implemented as part of multi-stage engine) as the new (V4) version of DataTable in the existing engine since it is an improved version of the existing DataTable